homework/src/index.js

```diff
@@ -32,13 +32,106 @@
 
   function main(url) {
     fetchJSON(url, (err, data) => {
+
+      //blue part of the page
       const root = document.getElementById('root');
+
+      createAndAppend('div', root, { id: 'myheading' });
+
+      const myHeading = document.getElementById('myheading');
+
+      createAndAppend('label', myHeading, { html: 'HYF Repositories ', class: 'rep-select' });
+
+      createAndAppend('select', myHeading, { class: 'rep-select', id: 'selectID' });
+
+      const select = document.getElementById('selectID');
+
       if (err) {
         createAndAppend('div', root, { html: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+
+        data.sort((a, b) => a.name.localeCompare(b.name));
+
+        data.forEach((obj) => { createAndAppend('option', select, { html: obj['name'], class: 'rep-names' }); });
+
+
+        //info-box of the page
+        function showDetails() {
```

Comment:
You should not normally define functions inside other functions, except when you dealing with relative short functions that you pass as a callback or event handler to some other function. 

You can define `showDetails()` at the file level, but you will need to pass the repository information that it needs to show as parameter to that function, e.g.:

```js
function showDetails(selectedRepo) {
  //...
}
```